person.rb

```diff
@@ -15,9 +18,13 @@ def can_use_services?
     false
   end
 
+  def validate_name
+    @name = @corrector.correct_name(@name)
+  end
+
   private
 
   def of_age?
-    return true if age >= 18
+    return true if @age >= 18
```

Comment:
I don't think you need the `@` on `name` because you have `attr_accessor :name` which creates a method behind the scenes that looks like this:

```ruby
def age
  @age
end
```